Sources/BSON/BSONBinary.swift

```diff
@@ -198,6 +198,21 @@ extension BSONBinary: BSONValue {
         }
     }
 
+    /// Converts this `BSONBinary` to a corresponding `JSON` in relaxed extendedJSON format.
+    internal func toRelaxedExtendedJSON() -> JSON {
+        self.toCanonicalExtendedJSON()
+    }
+
+    /// Converts this `BSONBinary` to a corresponding `JSON` in canonical extendedJSON format.
+    internal func toCanonicalExtendedJSON() -> JSON {
+        [
+            "$binary": [
+                "base64": .string(Data(self.data.readableBytesView).base64EncodedString()),
+                "subType": .string(self.subtype.rawValue.description)
```

Comment:
we need to ensure this subType is a hex string, so you need to use the `String` initializer that takes a radix, which in this case should be 16 for hex (https://developer.apple.com/documentation/swift/string/2997127-init)

Reply:
done